src/js/leaflet.storage.js

```diff
@@ -554,38 +554,55 @@ L.Storage.Map.include({
             shortUrl.value = this.options.shortUrl;
         }
         L.DomUtil.create('hr', '', container);
-        L.DomUtil.add('h4', '', container, L._('Download raw data'));
+        L.DomUtil.add('h4', '', container, L._('Download data'));
         var typeInput = L.DomUtil.create('select', '', container);
         typeInput.name = 'format';
-        L.DomUtil.add('small', 'help-text', container, L._('Only visible features will be downloaded.'));
+        var exportCaveat = L.DomUtil.add('small', 'help-text', container, L._('Only visible features will be downloaded.'));
+        exportCaveat.id = "export_caveat_text";
+        L.DomEvent.on(typeInput, 'change', function () {
+            if (typeInput.value === "umap") {
+                exportCaveat.style.display = "none";
+            } else {
+                exportCaveat.style.display = "inherit";
+            }
+        }, this);
         var types = {
             geojson: {
-                formatter: function (gj) {return JSON.stringify(gj, null, 2);},
+                formatter: function (map) {return JSON.stringify(map.toGeoJSON(), null, 2);},
                 ext: '.geojson',
                 filetype: 'application/json'
             },
             gpx: {
-                formatter: togpx,
+                formatter: function (gpx) {return togpx(map.toGeoJSON());},
                 ext: '.gpx',
                 filetype: 'application/xml'
             },
             kml: {
-                formatter: tokml,
+                formatter: function (map) {return tokml(map.toGeoJSON());},
                 ext: '.kml',
                 filetype: 'application/vnd.google-earth.kml+xml'
+            },
+            umap: {
+                name: L._('Raw uMap data'),
+                formatter: function (map) {return map.serialize();},
+                ext: '.umap',
+                filetype: 'application/json'
             }
         };
         for (var key in types) {
             if (types.hasOwnProperty(key)) {
                 option = L.DomUtil.create('option', '', typeInput);
                 option.value = option.innerHTML = key;
+                if (types[key].name) {
```

Comment:
I'd go for: `option.innerHTML = types[key].name || key;` (removing `innerHTML` from the line above).